app/app.go

```diff
@@ -144,15 +144,15 @@ func New(cfg *config.Config, databaseContext *dbaccess.DatabaseContext, interrup
 
 func (a *App) maybeSeedFromDNS() {
 	if !a.cfg.DisableDNSSeed {
-		if a.cfg.GRPCSeed != "" {
+		switch {
+		case a.cfg.GRPCSeed != "" && a.cfg.DNSSeed != "":
```

Comment:
Can you move this check to config/config.go where we validate command line arguments?